models/protected_tag.go

```diff
@@ -0,0 +1,118 @@
+// Copyright 2021 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package models
+
+import (
+	"strings"
+
+	"code.gitea.io/gitea/modules/base"
+	"code.gitea.io/gitea/modules/timeutil"
+
+	"github.com/gobwas/glob"
+)
+
+// ProtectedTag struct
+type ProtectedTag struct {
+	ID               int64     `xorm:"pk autoincr"`
+	RepoID           int64     `xorm:"UNIQUE(s)"`
+	NamePattern      string    `xorm:"UNIQUE(s)"`
+	NameGlob         glob.Glob `xorm:"-"`
+	WhitelistUserIDs []int64   `xorm:"JSON TEXT"`
+	WhitelistTeamIDs []int64   `xorm:"JSON TEXT"`
+
+	CreatedUnix timeutil.TimeStamp `xorm:"created"`
+	UpdatedUnix timeutil.TimeStamp `xorm:"updated"`
+}
+
+// BeforeInsert will be invoked by XORM before inserting a record
+func (pt *ProtectedTag) BeforeInsert() {
+	pt.CreatedUnix = timeutil.TimeStampNow()
+	pt.UpdatedUnix = timeutil.TimeStampNow()
+}
+
+// BeforeUpdate is invoked from XORM before updating this object.
+func (pt *ProtectedTag) BeforeUpdate() {
+	pt.UpdatedUnix = timeutil.TimeStampNow()
+}
```

Comment:
These two methods are unnecessary because xorm knows `created` and `updated` and will do that automatically.